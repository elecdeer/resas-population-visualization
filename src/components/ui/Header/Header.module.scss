@use "src/styles/variables";

.header {
  width: 100%;
  height: 48px;
  background-color: variables.$header-color;
  color: variables.$base-text-bk-color;
  text-align: center;
  margin: 0;

  h1 {
    margin: 0;
  }
}