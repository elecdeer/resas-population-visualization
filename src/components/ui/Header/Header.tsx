import React from "react";
import styles from "./Header.module.scss";

export type HeaderProps = {
  pageTitle: string;
};

export const Header: React.FC<HeaderProps> = ({ pageTitle, children }) => {
  return (
    <header className={styles["header"]}>
      <h1>{children ?? pageTitle}</h1>
    </header>
  );
};
