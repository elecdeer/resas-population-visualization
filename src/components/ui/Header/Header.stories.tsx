import { Header, HeaderProps } from "./Header";

export default {
  title: "ui/Header",
  component: Header,
};

export const Default = (args: HeaderProps) => (
  <Header {...args} pageTitle={"Title"} />
);
