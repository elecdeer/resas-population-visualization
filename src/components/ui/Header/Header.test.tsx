import { Header } from "./Header";
import { render } from "@testing-library/react";

test("Header Component Snapshot Test", () => {
  const { asFragment } = render(<Header pageTitle={"Title"} />);
  expect(asFragment()).toMatchSnapshot();
});
