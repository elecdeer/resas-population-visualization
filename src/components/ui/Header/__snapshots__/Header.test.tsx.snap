// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Snapshot Test 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <h1>
      Title
    </h1>
  </div>
</DocumentFragment>
`;
